net/pfSense-pkg-freeradius2/files/usr/local/pkg/freeradius.inc

```diff
@@ -4409,4 +4415,834 @@ EOD;
+			$input_errors[] = "The 'Invalid Password Attempts' field should contain only sane secure values. Values higher than 20 are not allowed.";
+		}
+	}
+
+	// Token Password Length
+	if ($post['varsettingsmotptokenlength'] != '') {
+		if (!preg_match('/^[0-9]{1,2}(?:[-][0-9]{1,2}){1}$/i', $post['varsettingsmotptokenlength'])) {
+			$input_errors[] = "The 'Token Password Length' field must contain a valid integer range between 1 and 16, separated with a '-' character; e.g.: 1-6.";
+		} elseif (!is_intrange($post['varsettingsmotptokenlength'], 1, 16)) {
+			$input_errors[] = "The 'Token Password Length' field must contain a valid integer range between 1 and 16, separated with a '-' character; e.g.: 1-6.";
+		}
+	}
+
+	/*
+	 * TODO:
+	 * Logging Configuration - Additional Information for Bad Attempts, Additional Information for Good Attempts
+	 * Miscellaneous Configuration - Default Profile, Profile Attribute, Access Attribute
+	 * Group Membership Options - Groupname Attribute, Group Membership Filter, Group Membership Attribute
+	 */
+
+}
+
+/* EAP settings input validation */
+function freeradius_validate_eap($post, &$input_errors) {
+
+	// Disable Weak EAP Types sanity check
+	if ($post['vareapconfdisableweakeaptypes'] == 'on') {
+		if ($post['vareapconfdefaulteaptype'] == 'md5' || $post['vareapconfdefaulteaptype'] == 'gtc' || $post['vareapconfdefaulteaptype'] == 'leap') {
+			$input_errors[] = "The 'Default EAP Type' field must not contain any of the weak types when 'Disable Weak EAP Types' is enabled.";
+		}
+	}
+
+	// EAP Maximum Sessions Tracking Per Server
+	if ($post['vareapconfmaxsessions'] != '' && !is_numericint($post['vareapconfmaxsessions'])) {
+		$input_errors[] = "The 'Maximum Sessions Tracking Per Server' field must contain an integer value.";
+	}
+
+	// EAP Certificates for TLS sanity checking
+	if ($post['vareapconfchoosecertmanager'] == 'on') {
+		// EAP Private Key Password should be empty if using pfSense Cert Manager
+		if ($post['vareapconfprivatekeypassword'] != '') {
+			$input_errors[] = "When using pfSense Certificate Management, the 'Private Key Password' field must be empty.";
+		}
+	} else {
+		// SSL CA Certificate, SSL Revocation List and SSL Server Certificate should be 'none' if using the deprecated FreeRADIUS Cert Manager
+		if ($post['ssl_ca_cert'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL CA Certificate' field must be set to 'none'.";
+		}
+		if ($post['ssl_ca_crl'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL Revocation List' field must be set to 'none'.";
+		}
+		if ($post['ssl_server_cert'] != 'none') {
+			$input_errors[] = "When using the deprecated FreeRADIUS Cert Manager, the 'SSL Server Certificate' field must be set to 'none'.";
+		}
+	}
+
+	// EAP-TLS Fragment Size
+	if ($post['vareapconffragmentsize'] != '') {
+		if (!is_numericint($post['vareapconffragmentsize'])) {
```

Comment:
I'd combine the test and print the same error for both, mentioning the type and limit.